tjenestebuss-integrasjon/src/main/kotlin/no/nav/pensjon/brev/tjenestebuss/tjenestebussintegrasjon/services/arkiv/ArkivTjenestebussService.kt

```diff
@@ -0,0 +1,90 @@
+            val response = arkivClient.bestillBrev(BestillBrevRequest().apply {
+                brevKode = bestillBrevRequestDto.brevKode
+                brevGruppe = bestillBrevRequestDto.brevGruppe
+                redigerbart =
+                    JAXBElement(QName("redigerbart"), Boolean::class.java, bestillBrevRequestDto.isRedigerbart)
+                sprakKode = bestillBrevRequestDto.sprakkode
+                sakskontekst = Sakskontekst().apply {
+                    saksbehandlernavn = bestillBrevRequestDto.sakskontekstDto.saksbehandlernavn
+                    saksbehandlerId = bestillBrevRequestDto.sakskontekstDto.saksbehandlerId
+                    journalenhet = bestillBrevRequestDto.sakskontekstDto.journalenhet
+                    gjelder = bestillBrevRequestDto.sakskontekstDto.gjelder
+                    dokumenttype = bestillBrevRequestDto.sakskontekstDto.dokumenttype
+                    dokumentdato = bestillBrevRequestDto.sakskontekstDto.dokumentdato
+                    fagsystem = bestillBrevRequestDto.sakskontekstDto.fagsystem
+                    fagomradeKode = bestillBrevRequestDto.sakskontekstDto.fagomradekode
+                    innhold = bestillBrevRequestDto.sakskontekstDto.innhold
+                    kategori = bestillBrevRequestDto.sakskontekstDto.kategori
+                    saksid = bestillBrevRequestDto.sakskontekstDto.saksid
+                    sensitivitetsgrad = bestillBrevRequestDto.sakskontekstDto.sensitivitet
+                }
+            })
+            logger.info("Opprettet brev med journalpostId: ${response!!.journalpostId} i sakId: ${bestillBrevRequestDto.sakskontekstDto.saksid} ")
+            return BestillBrevResponseDto.Journalpost(response.journalpostId)
+        } catch (ex: BestillBrevOpprettelseJournalpostFeilet) {
+            logger.error("En feil oppstod under opprettelse av journalpost: ${maskerFnr(ex.faultInfo.errorMessage)}")
+            return BestillBrevResponseDto.Failure(ex.faultInfo)
+        } catch (ex: BestillBrevHenteBrevdataFeilet) {
+            logger.error("En feil oppstod under henting av brevdata: ${maskerFnr(ex.faultInfo.errorMessage)}")
+            return BestillBrevResponseDto.Failure(ex.faultInfo)
+        } catch (ex: BestillBrevManglerObligatoriskInput) {
+            logger.error("En feil oppstod under opprettelse av brev, mangler obligatoriske felter: ${maskerFnr(ex.faultInfo.errorMessage)}")
+            return BestillBrevResponseDto.Failure(ex.faultInfo)
+        } catch (ex: BestillBrevAdresseIkkeRegistrert) {
+            logger.error("En feil oppstod under opprettelse av brev, adresse ikke registrert: ${maskerFnr(ex.faultInfo.errorMessage)}")
+            return BestillBrevResponseDto.Failure(ex.faultInfo)
+        }
+    }
+}
+
+sealed class BestillBrevResponseDto {
+    data class Journalpost(val journalpostId: String) : BestillBrevResponseDto()
+    data class Failure(private val fault: no.nav.virksomhet.tjenester.felles.v1.StelvioFault) :
```

Comment:
Ikke et veldig stort problem, men feltene du deklarerer i klasse-kroppen vil ikke bli inkludert i equals/hashcode, og fault brukes ikke til noe etter konstruksjon siden du bruker den til å initiere feltene i klasse-kroppen. Jeg ville nok heller deklarert feltene i primær-konstruktøren, også laget en sekundær-konstruktør eller statisk-funksjon for å lage Failure basert på StelvioFault.

Reply:
Det er mulig at vi kanskje ønsker å kunne reagere forskjellig på f.eks ManglerObligatoriskInput eller AdresseIkkeRegistrert, så vi kan fort ha behov for forskjellige klasser for de feilene.